---
title: "Differential Expression"
author: "`r getOption('author')`"
date: "`r Sys.Date()`"
bibliography: bibliography.bib
params:
    seurat: "data/bcbFilteredSeuratSubset.rda"
    intgroup: "genotype"
    contrast: !r c(numerator = "Het", denominator = "WT")
---

```{r setup, cache=FALSE, message=FALSE, warning=FALSE}
library(Seurat)
library(bcbioSingleCell)
library(tidyverse)
source("setup.R")
seuratName <- load(params$seurat)
seurat <- get(seuratName, inherits = FALSE)
rm(list = seuratName)
knitr::opts_chunk$set(cache = FALSE)
```

```{r header, child="_header.Rmd", eval=file.exists("_header.Rmd")}
```



Identify differentially expressed genes between two groups of cells using the LRT model proposed in McDavid et al, Bioinformatics, 2013.

```{r cells_per_cluster}
# unique(seurat@meta.data[[params$intgroup]])
ident <- levels(seurat@ident)
cells_per_cluster <- lapply(seq_along(ident), function(a) {
    cells <- seurat@ident %>%
        .[. == ident[[a]]] %>%
        names
    meta <- seurat@meta.data[cells, ]
    # Cell groupings to use for differential expression
    cells1 <- meta %>%
        .[.[[params$intgroup]] == params$contrast[[1]], ] %>%
        rownames %>%
        sort %>%
        WhichCells(seurat, cells.use = ., ident = ident)
    cells2 <- meta %>%
        .[.[[params$intgroup]] == params$contrast[[2]], ] %>%
        rownames %>%
        sort %>%
        WhichCells(seurat, cells.use = ., ident = ident)
    list(cells1 = cells1,
         cells2 = cells2)
}) %>%
    set_names(ident)
saveData(cells_per_cluster)
```

Testing function priority:

- `DiffExpTest()` (default; aka "bimod" in FindMarkers). This takes about 2 hours per cluster.
- `TobitTest()`. This seems to either be very slow to load or not report progress.
- `NegBinomDETest()` (UMI only). This takes about 20 minutes per cluster.

Additional possible tests (not run):

- `NegBinomRegDETest()` (UMI only).
- `PoissonDETest()` (UMI only).
- `DiffTTest()` (don't use).

Decided to settle on only using `NegBinomDETest()` at the moment.

```{r run_de}
# Double check that numerator and denominator are set correctly. This isn't clear in the current Seurat documentation. It appears that `cells.1` is the numerator.
differential_expression <-
    pbapply::pblapply(seq_along(cells_per_cluster), function(a) {
        cells1 <- cells_per_cluster[[a]]$cells1
        cells2 <- cells_per_cluster[[a]]$cells2
        suppressWarnings(
            NegBinomDETest(seurat, cells.1 = cells1, cells.2 = cells2)
        )
    }) %>%
    set_names(names(cells_per_cluster))
saveData(differential_expression)
```



```{r footer, child="_footer.Rmd", eval=file.exists("_footer.Rmd")}
```
